src/experimental/rip7560/accounts/implementations/toSimpleNativeSmartAccount.ts

```diff
@@ -0,0 +1,692 @@
+
+    async sign(parameters) {
+      const address = await this.getAddress()
+
+      const hash = toReplaySafeHash({
+        address,
+        chainId: client.chain!.id,
+        hash: parameters.hash,
+      })
+
+      const signature = await sign({ hash, owner })
+
+      return signature
+    },
+
+    async signMessage(parameters) {
+      const { message } = parameters
+      const address = await this.getAddress()
+
+      const hash = toReplaySafeHash({
+        address,
+        chainId: client.chain!.id,
+        hash: hashMessage(message),
+      })
+
+      const signature = await sign({ hash, owner })
+
+      return signature
+    },
+
+    async signTypedData(parameters) {
+      const { domain, types, primaryType, message } =
+        parameters as TypedDataDefinition<TypedData, string>
+      const address = await this.getAddress()
+
+      const hash = toReplaySafeHash({
+        address,
+        chainId: client.chain!.id,
+        hash: hashTypedData({
+          domain,
+          message,
+          primaryType,
+          types,
+        }),
+      })
+
+      const signature = await sign({ hash, owner })
+
+      return signature
+    },
+
+    async signTransaction(parameters) {
+      const { chainId = client.chain!.id, ...transaction } = parameters
+
+      const address = await this.getAddress()
+      const serializableTransaction = {
+        ...transaction,
+        sender: address,
+        chainId,
+      } as TransactionSerializable
+      const hash = keccak256(serializeTransaction(serializableTransaction))
+
+      const signature = await sign({ hash, owner })
+
+      return signature
+    },
+
+    transaction: {
+      async estimateGas(transaction) {
+        return {
+          verificationGasLimit: BigInt(
+            Number(transaction.verificationGasLimit ?? 0n),
+          ),
+        }
+      },
+    },
+  })
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+// Utilities
+/////////////////////////////////////////////////////////////////////////////////////////////
+
+/** @internal */
+export async function sign({
+  hash,
+  owner,
+}: { hash: Hash; owner: LocalAccount }) {
+  if (owner.sign) return owner.sign({ hash })
+
+  throw new BaseError('`owner` does not support raw sign.')
+}
+
+/** @internal */
+export function toReplaySafeHash({
+  address,
+  chainId,
+  hash,
+}: { address: Address; chainId: number; hash: Hash }) {
+  return hashTypedData({
+    domain: {
+      chainId,
+      name: 'Simple Native Smart Wallet',
+      verifyingContract: address,
+      version: '1',
+    },
+    types: {
+      SimpleNativeSmartWalletMessage: [
+        {
+          name: 'hash',
+          type: 'bytes32',
+        },
+      ],
+    },
+    primaryType: 'SimpleNativeSmartWalletMessage',
+    message: {
+      hash,
+    },
+  })
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////////
+// Constants
```

Comment:
I think it would look better if you separate it into a separate file such as constants/abis.ts.

Reply:
I was sticking to the format of files that were implemented in `account-abstraction` folder, so I think this time we can just let it be like this, and talk to wevm team about this (or just making PRs on their repo).